assets/javascripts/discourse/components/campaign-banner.js.es6

```diff
@@ -0,0 +1,143 @@
+import { action } from "@ember/object";
+import Component from "@ember/component";
+import discourseComputed, { observes } from "discourse-common/utils/decorators";
+import { inject as service } from "@ember/service";
+import User from "discourse/models/user";
+import { Promise } from "rsvp";
+
+export default Component.extend({
+  router: service(),
+  dismissed: false,
+  loading: false,
+  showContributors: false,
+  contributors: [],
+  classNameBindings: [
+    "isSidebar:campaign-banner-sidebar",
+    "shouldShow:campaign-banner",
+  ],
+
+  init() {
+    this._super(...arguments);
+    const dismissed = document.cookie.includes(
```

Comment:
Can we use the `KeyValueStore` like the global banner users? It's backed by localStorage.

Reply:
Sure -- I'll look into that. Didn't know it was an option :D